### PR TITLE
Refactor Auth and display frontend edit links

### PR DESCRIPTION
**Pull request by jsturek**
May test on https://events-dev.unl.edu

* Replace old CAS with newer one
* Display event edit button on frontend when user logged in and has edit permission

@skoolbus39 - need to verify edit button style and placement.

### CODE REVIEW

**Comment by jsturek** [comment]
@ericras and @skoolbus39 This is now ready for review and may be tested on https://events-dev.unl.edu.

File: src/UNL/UCBCN/Manager/AuthAction.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            throw new NotFoundException('Not Found');
+        }
+    }
+
+    private function logout() {
+        $auth = new Auth();
+        $auth->logout();
+        die();
+    }
+}
```
Comment (skoolbus39) [inline src/UNL/UCBCN/Manager/AuthAction.php:27]: No newline at end of file

**Review by skoolbus39 — CHANGES_REQUESTED** [review]
Looks good besides the no newline at end of file

File: src/UNL/UCBCN/Manager/AuthAction.php (hunk 1 above)
Comment (jsturek) [reply src/UNL/UCBCN/Manager/AuthAction.php]: done

**Review by skoolbus39 — APPROVED** [review]

**Comment by ericras** [comment]
https://events-dev.unl.edu/manager/https/manager/logout

Fatal error: Uncaught Error: Class name must be a valid object or a string in /var/www/html/events-dev.unl.edu/src/UNL/UCBCN/Manager/Controller.php:100 Stack trace: #0 /var/www/html/events-dev.unl.edu/src/UNL/UCBCN/Manager/Controller.php(50): UNL\UCBCN\Manager\Controller->run() #1 /var/www/html/events-dev.unl.edu/www/manager/index.php(25): UNL\UCBCN\Manager\Controller->__construct(Array) #2 {main} thrown in /var/www/html/events-dev.unl.edu/src/UNL/UCBCN/Manager/Controller.php on line 100

File: www/templates/default/html/manager/navigation.tpl.php [hunk 2]
```diff
@@ -27,6 +27,10 @@
             <?php if ($user->hasPermission(\UNL\UCBCN\Permission::CALENDAR_EDIT_PERMISSIONS_ID, $calendar->id)): ?>
                 <li><a class="dcf-txt-decor-hover" href="<?php echo $calendar->getUsersURL() ?>">Users &amp; Permissions</a></li>
             <?php endif; ?>
+
+            <?php if ($user->hasPermission(\UNL\UCBCN\Permission::CALENDAR_EDIT_PERMISSIONS_ID, $calendar->id)): ?>
+                <li><a class="dcf-txt-decor-hover" href="/manager/logout">Logout</a></li>
+            <?php endif; ?>
```
Comment (ericras) [inline www/templates/default/html/manager/navigation.tpl.php:33]: Prepend Util::getBaseURL() on this

File: www/templates/default/html/manager/Controller.tpl.php [hunk 3]
```diff
@@ -40,6 +41,7 @@
 $page->addStyleDeclaration("#dcf-mobile-toggle-menu {display: none!important}");
 
 //javascript
+$page->addScriptDeclaration('WDN.setPluginParam("idm", "logout", "' .  Util::getBaseURL() . '/manager/logout");');
```
Comment (ericras) [inline www/templates/default/html/manager/Controller.tpl.php:44]: Somehow for this we're getting "https://events-dev.unl.edu/manager/https/manager/logout"

File: www/templates/default/html/Controller.tpl.php [hunk 4]
```diff
@@ -39,6 +40,7 @@
 $page->addStyleDeclaration("#dcf-mobile-toggle-menu {display: none!important}");
 
 //javascript
+$page->addScriptDeclaration('WDN.setPluginParam("idm", "logout", "' . Util::getBaseURL() . '/manager/logout");');
```
Comment (ericras) [inline www/templates/default/html/Controller.tpl.php:43]: Not generating the correct logout link. I'm getting "https://events-dev.unl.edu/TractorSafetyTraining/https/manager/logout"

File: www/templates/default/html/manager/Controller.tpl.php [hunk 5]
```diff
@@ -41,7 +41,7 @@
 $page->addStyleDeclaration("#dcf-mobile-toggle-menu {display: none!important}");
 
 //javascript
-$page->addScriptDeclaration('WDN.setPluginParam("idm", "logout", "' .  Util::getBaseURL() . '/manager/logout");');
+$page->addScriptDeclaration('WDN.setPluginParam("idm", "logout", "logout");');
```
Comment (ericras) [inline www/templates/default/html/manager/Controller.tpl.php:44]: Still does https://events-dev.unl.edu/manager/TractorSafetyTraining/logout on the manager side.  Probably still need to prepend a working base url string

**Comment by jsturek** [comment]
@ericras ready for review

**Review by ericras — APPROVED** [review]

**Pull request merged**